src/Avalonia.Base/IAvaloniaObject.cs

```diff
@@ -41,6 +41,19 @@ public interface IAvaloniaObject
         /// <returns>The value.</returns>
         T GetValue<T>(DirectPropertyBase<T> property);
 
+        /// <summary>
+        /// Gets an <see cref="AvaloniaProperty"/> base value.
+        /// </summary>
+        /// <typeparam name="T">The type of the property.</typeparam>
+        /// <param name="property">The property.</param>
+        /// <param name="maxPriority">The maximum priority for the value.</param>
+        /// <remarks>
+        /// Gets the value of the property, if set on this object with a priority equal or lower to
+        /// <paramref name="maxPriority"/>, otherwise <see cref="Optional{T}.Empty"/>. Note that
+        /// this method does not return property values that come from inherited or default values.
+        /// </remarks>
+        Optional<T> GetBaseValue<T>(StyledPropertyBase<T> property, BindingPriority maxPriority);
```

Comment:
The name doesn't feel natural. `GetValueByPriority` ?

Reply:
The naming comes from WPF: [`GetAnimationBaseValue`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.getanimationbasevalue?view=netcore-3.1) except that it now takes a binding priority so I removed the "Animation" part from the name.

`GetValueByPriority` might be a good alternative though.

Reply:
For the record: we decided to add a `minPriority` parameter to this method and change its name to `GetValueByPriority` in a future PR.